File: 15174-1026-13-roman-to-integer/15174-1026-13-roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        HashMap<String, Integer> romanMap = new HashMap<>();        
        romanMap.put("I", 1);
        romanMap.put("V", 5);
        romanMap.put("X", 10);
        romanMap.put("L", 50);
        romanMap.put("C", 100);
        romanMap.put("D", 500);
        romanMap.put("M", 1000);
        
        String[] str = s.split("");
        int number = 0;
        
        for (int i = 0; i < str.length; i++) {          
                String currentChar = str[i];
            
                String nextChar = "";

                if (i < str.length - 1) {
                    nextChar = str[i + 1];
                }
            
                String prevChar = "";
                
                if (i != 0) {
                    prevChar = str[i - 1];
                }
                
                if (nextChar != "" && (romanMap.get(currentChar) < romanMap.get(nextChar))) {
                    if (prevChar != "" && (romanMap.get(currentChar) > romanMap.get(currentChar))) {
                        continue;
                    }
                    
                    number += romanMap.get(nextChar) - romanMap.get(currentChar);
                } else {
                    if (prevChar != "" && (romanMap.get(currentChar) > romanMap.get(prevChar))) {
                        continue;
                    }
                    
                    number += romanMap.get(currentChar);
                }            
        }
        
        return number;
    }
}

File: 15174-1026-13-roman-to-integer/15174-1026-13-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        romanDict = {"I": 1, "V": 5, "X": 10, "L": 50, "C": 100, "D": 500, "M": 1000}

        num = 0
        for i in range(len(s)):
            currentChar = s[i]
            try:
                nextChar = s[i + 1]
            except IndexError:
                nextChar = None

            if i == 0:
                prevChar = None
            else:
                prevChar = s[i - 1]

            if nextChar and (romanDict[currentChar] < romanDict[nextChar]):
                if prevChar and (romanDict[currentChar] > romanDict[prevChar]):
                    continue
                num = num + (romanDict[nextChar] - romanDict[currentChar])
            else:
                if prevChar and (romanDict[currentChar] > romanDict[prevChar]):
                    continue
                num = num + romanDict[currentChar]

        return num
